src/api/posts-api.js

```diff
@@ -1,10 +1,32 @@
 import axios from 'axios';
 import store from '../store';
-import { getPostsSuccess } from 'actions/post-actions';
+import { getPostsSuccess, getPostsWaiting, postSuccess, postWaiting } from 'actions/post-actions';
 
 export const getPosts = () => {
-    return axios.get('../posts.json', {responseType: 'json'})
+    return axios.get('../posts.json', { responseType: 'json' })
         .then(response => {
-            store.dispatch(getPostsSuccess(response.data.posts))
+            console.log('Waiting for data...');
+            store.dispatch(getPostsWaiting());
```

Comment:
You should pass this method to components using connect, as a result you will not need to use store to dispatch action